engine/crates/composition/src/emit_federated_graph/context.rs

```diff
@@ -54,3 +50,9 @@ impl<'a> Context<'a> {
             .push(federated::InterfaceField { interface_id, field_id });
     }
 }
+
+impl Drop for Context<'_> {
+    fn drop(&mut self) {
+        self.out.strings = std::mem::take(&mut self.strings_ir).into_federated_strings();
```

Comment:
We could assert that `self.out.strings` is empty at this point, but since the drop impls run during unwinding, it would be a panic in a panic, and these are terrible.

Other possible solutions:

- Regular `finalize()` method instead of Drop.
- Put the IndexSet directly inside `FederatedGraph`. It should be backwards compatible, and there would be no more room for error.